### PR TITLE
Arrow 10.0.0 support

### PR DESCRIPTION
**Pull request by Dakostu**
Support building VAST against Arrow 10.0.0.

### :memo: Reviewer Checklist

Review this pull request by ensuring the following items:

- [ ] All user-facing changes have changelog entries
- [ ] User-facing changes are reflected on [vast.io](https://github.com/tenzir/vast/tree/master/web)

### CODE REVIEW

File: changelog/unreleased/changes/2685--update-arrow-to-10.0.0.md [hunk 1]
```diff
@@ -0,0 +1 @@
+Building VAST from source now requires Arrow 10.0.0 (https://arrow.apache.org/blog/2022/10/31/10.0.0-release/).
```
Comment (dominiklohmann) [inline changelog/unreleased/changes/2685--update-arrow-to-10.0.0.md:1]:
```suggestion
Building VAST from source now requires [Apache Arrow 10.0](https://arrow.apache.org/blog/2022/10/31/10.0.0-release/) or newer.
```

**Review by dominiklohmann — APPROVED** [review]
Just one minor request for the changelog entry. We also need to bump nixpkgs (@tobim), and then I think this can be shipped.

File: web/docs/setup/build.md [hunk 2]
```diff
@@ -29,7 +29,7 @@ dependencies and versions.
 |✓|[CMake](https://cmake.org)|>= 3.19|Cross-platform tool for building, testing and packaging software.|
 |✓|[CAF](https://github.com/actor-framework/actor-framework)|>= 0.17.6|Implementation of the actor model in C++. (Bundled as submodule.)|
 |✓|[FlatBuffers](https://google.github.io/flatbuffers/)|>= 1.12.0|Memory-efficient cross-platform serialization library.|
-|✓|[Apache Arrow](https://arrow.apache.org)|>= 8.0.0|Required for in-memory data representation. Must be built with Compute, Zstd and Parquet enabled.|
+|✓|[Apache Arrow](https://arrow.apache.org)|>= 10.0.0|Required for in-memory data representation. Must be built with Compute, Zstd and Parquet enabled.|
```
Comment (Dakostu) [inline web/docs/setup/build.md:32]: Can this doc change be reverted or are we "officially" supporting Arrow >=10.0.0 now?

**Review by Dakostu — COMMENTED** [review]
@tobim So I ran Cmake on this this branch using an Arrow-9-based and an Arrow-10-based machine and they both succeeded. Building w/ Arrow 9 worked, I'm still building w/ Arrow 10, gonna assume both builds will run correctly...
I did leave one question though.

File: web/docs/setup/build.md (hunk 2 above)
Comment (tobim) [reply web/docs/setup/build.md]: Good observation! I reverted it now.

**Review by tobim — APPROVED** [review]

**Pull request merged**